docs/_ext/scylladb_cc_properties.py

```diff
@@ -0,0 +1,53 @@
+import os
+import re
+import yaml
+from sphinx.application import Sphinx
+
+CONFIG_FILE_PATH = '../db/config.cc'
+DESTINATION_PATH = '_data/db_config.yaml'
+
+def create_yaml_file(destination, data):
+    current_data = None
+    
+    try:
+        with open(destination, 'r') as file:
+            current_data = yaml.safe_load(file)
+    except FileNotFoundError:
+        pass
+
+    if current_data != data:
+        os.makedirs(os.path.dirname(destination), exist_ok=True)
+        with open(destination, 'w') as file:
+            yaml.dump(data, file)
+
+def parse_db_properties(config_file):
+    with open(config_file, 'r') as file:
+        content = file.read()
+
+    pattern = r',\s*(\w+)\(this,\s*"(\w+)",\s*(?:liveness::(\w+),\s*)?value_status::(\w+),\s*([^,]+),\s*"([^"]+)"\)'
+    matches = re.findall(pattern, content)
+
+    properties = []
+    for match in matches:
+        property_data = {
+            "name": match[1].strip(),
+            "value_status": match[3].strip(), 
+            "default": match[4].strip(),
+            "liveness": 'True' if match[2] else 'False',
+            "description": match[5].strip().replace('\n', '<br />')
+        }
+        properties.append(property_data)
+
+    return properties
+
+def generate_cc_docs(app: Sphinx):
+    dest_path = os.path.join(app.builder.srcdir, DESTINATION_PATH)
+    parsed_properties = parse_db_properties(CONFIG_FILE_PATH)
+    create_yaml_file(dest_path, parsed_properties)
+
+def note_dependency(app: Sphinx, docname, source):
+    app.env.note_dependency(CONFIG_FILE_PATH)
```

Comment:
unfortunately, unlike a `SphinxDirective`, which has its "current document", we don't have access to it. in other words, one cannot note the dependency in `setup()` of an extension. the `datatemplate` directive, however does this for us, but it adds the dependency to the file specified with the "source" option, which is `../_data/db_config.yaml`. so whenever the yaml file is updated, sphinx rebuilds `db-config.rst`. but the change from .cc file is not able to be populated to `db_config.yaml`.

not to mention how to teach sphinx what to do to "build" the document. we just add a hook to generate the yaml file when builder is initialized.

sorry, i didn't realize that it was not a directive.

if we want to stick with sphinxcontrib.datatemplates, guess there is no simple way to teach the dependency to sphinx.